### ISSUE TITLE
Accessing the required property of references

### ISSUE THREAD

**Issue by knellr** [mattpolzin/OpenAPIKit#220]
Hi, I'm trying out OpenAPIKit for code generation purposes. 

Working with JSON schemas, the `required` property doesn't appear to respect an object's required property array when encountering references in `properties(takingRequirementsFrom:)`. This information seems to get lost, since the original array is never stored, and references just return `true` for the `required` property.

Is it correct in the OpenAPI spec that the required properties array shouldn't apply to references? Or is there a way in OpenAPIKit to access the `required` property for references?

**Comment by mattpolzin** [comment]
The inability to represent optionality of references is a shortcoming (I’d even go so far as to call it a bug) of the latest OpenAPIKit release as of now.

A fix has been implemented, but it is a breaking change so it is being held for the next major version of OpenAPIKit. I intend to release an _alpha_ of that next version today, actually. This first alpha release will be fairly safe to use for OpenAPI v3.0.x integrations — it will see more changes in its support of OpenAPI v3.1.x

In short: if you’re willing to try out an alpha (shouldn’t give you any problems unless you’re after the newest version of the OpenAPI spec that is not used much yet), I’ll reply to this thread with instructions after I’ve written up release notes and published a new release.

**Comment by knellr** [comment]
Thanks for the quick response! That sounds promising, and I'll definitely give the alpha a try when it's up.

**Comment by mattpolzin** [comment]
Ok, [v3.0.0-alpha.1](https://github.com/mattpolzin/OpenAPIKit/releases/tag/3.0.0-alpha.1) has been released.

You almost certainly want to stick with OpenAPI specification vv3.0, both because the 3.1 version is not used hardly anywhere in the wild that I've seen and because OpenAPIKit support for it is a bit preliminary so far).

For that, you can mostly skim the release notes and focus on the section titled "V3.0 of the specification" about half way down. You will need to modify package manifest dependency descriptions and `import` lines according to those release notes.

**Comment by mattpolzin** [comment]
I'm closing this now that a fix is available in the latest alpha release. If you have any more questions or the fix isn't solving your use case please still follow up here!
